modules/navier_stokes/src/bcs/INSADDisplaceBoundaryBC.C

```diff
@@ -18,17 +18,23 @@ INSADDisplaceBoundaryBC::validParams()
 {
   InputParameters params = ADNodalBC::validParams();
   params.addClassDescription("Boundary condition for displacing a boundary");
-  params.addRequiredCoupledVar("velocity", "The velocity at which to displace");
+  params.addRequiredParam<MooseFunctorName>("velocity", "The velocity at which to displace");
   params.addRequiredParam<unsigned short>(
       "component", "What component of velocity/displacement this object is acting on.");
+  params.addRequiredParam<SubdomainName>(
+      "associated_subdomain",
```

Comment:
this could be limiting in the futue, add a node to split the nodeset if the subdomain dont extend to the entire nodeset?

Reply:
I'm not quite sure what you mean in your second clause